wso2.org.component.javaparser/src/main/java/component/javaparser/WSO2POMReader.java

```diff
@@ -0,0 +1,293 @@
+        for (int i = 0; i < nlist.getLength(); i++) {
+            ele = (Element) nlist.item(i);
+            Element dependecyVersion = doc.createElement("apache.felix.scr.ds.annotations.version");
+            dependecyVersion.appendChild(doc.createTextNode("1.2.4"));
+            ele.appendChild(dependecyVersion);
+        }
+    }
+
+    private void addSCRDependencyToComponent(Document doc, NodeList nlist) {
+        Element ele = null;
+        for (int i = 0; i < nlist.getLength(); i++) {
+            ele = (Element) nlist.item(i);
+            Element dependecy = doc.createElement("dependency");
+            Element groupId = doc.createElement("groupId");
+            groupId.appendChild(doc.createTextNode("org.apache.felix"));
+            Element artifactId = doc.createElement("artifactId");
+            artifactId.appendChild(doc.createTextNode("org.apache.felix.scr.ds-annotations"));
+            Element scope = doc.createElement("scope");
+            scope.appendChild(doc.createTextNode("provided"));
+            ele.appendChild(dependecy);
+            dependecy.appendChild(groupId);
+            dependecy.appendChild(artifactId);
+            dependecy.appendChild(scope);
+        }
+    }
+
+    private boolean hasSCRPluginNode(NodeList plugins) {
+        boolean isSCR = false;
+        Element plu = null;
+        for (int i = 0; i < plugins.getLength(); i++) {
+            plu = (Element) plugins.item(i);
+            String name = plu.getElementsByTagName("artifactId").item(0).getTextContent().toString();
+            if (name.equals("maven-scr-plugin")) {
+                isSCR = true;
+            }
+        }
+        return isSCR;
+    }
+
+    public void addSCRdependecyToParent(Document doc, NodeList nlist) {
+        Element ele = null;
+        Element emp = null;
+        for (int i = 0; i < nlist.getLength(); i++) {
+            emp = (Element) nlist.item(i);
+            Element dependecy = doc.createElement("dependency");
+            Element groupId = doc.createElement("groupId");
+            groupId.appendChild(doc.createTextNode("org.apache.felix"));
+            Element artifactId = doc.createElement("artifactId");
+            artifactId.appendChild(doc.createTextNode("org.apache.felix.scr.ds-annotations"));
+            Element version = doc.createElement("version");
+            version.appendChild(doc.createTextNode("${apache.felix.scr.ds.annotations.version}"));
+            emp.appendChild(dependecy);
+            dependecy.appendChild(groupId);
+            dependecy.appendChild(artifactId);
+            dependecy.appendChild(version);
+        }
+    }
+
+    public void updatePomFile(Document doc, String file) throws TransformerException, IOException {
+        doc.setXmlStandalone(true);
+        doc.getDocumentElement().normalize();
+        TransformerFactory transformerFactory = TransformerFactory.newInstance();
+            Transformer transformer = transformerFactory.newTransformer();
```

Comment:
Please format the above code. Checkstyle plugin can be run from IDE against the modified source files to find formatting issues in your code.

Reply:
fixed checkstyle.